### DISCUSSION TITLE
`partialObject` warning in Rollup

### DISCUSSION THREAD

**Discussion by midnight-wonderer** [ramda/ramda#3483] [General]
I used [rollup.js](https://rollupjs.org/) to bundle ramda, this lead to the following warning:

~~~
(!) "this" has been rewritten to "undefined"
https://rollupjs.org/troubleshooting/#error-this-is-undefined
node_modules/ramda/es/partialObject.js
30:  * @symb R.partialObject(f, { a, b })({ c, d }) = f({ a, b, c, d })
31:  */
32: var partialObject = /*#__PURE__*/_curry2((f, o) => props => f.call(this, mergeDeepRight(o, props)));
                                                                       ^
33: export default partialObject;
~~~

Basically, as stated in the first line: `"this" has been rewritten to "undefined"`.  
Can we improve this so Rollup won't complain?

**Comment by kedashoe** [comment]
ty for the issue @midnight-wonderer , looks like a bug on our part. `partialObject` [is defined with](https://github.com/ramda/ramda/blob/v0.30.1/source/partialObject.js#L32) arrow functions, so there is no `this`. We just need to update it to use good old `function`, eg [how call is defined](https://github.com/ramda/ramda/blob/v0.30.1/source/call.js#L41) (and many other examples).

Would you like to submit a PR?

**Reply by midnight-wonderer** [reply]
Why not; I just wonder why we need `call` at all, so I started another discussion in the PR:https://github.com/ramda/ramda/pull/3484.
